libcontainer/cgroups/utils.go

```diff
@@ -494,6 +494,21 @@ func getHugePageSizeFromFilenames(fileNames []string) ([]string, error) {
 	return pageSizes, nil
 }
 
+// WalkCgroups calls walkFn on the cgroup directory at path and on all its
+// subcgroups.
+func WalkCgroups(path string, walkFn func(path string) error) error {
+	return filepath.Walk(path, func(p string, info os.FileInfo, iErr error) error {
+		dir, file := filepath.Split(p)
```

Comment:
This is also very expensive, especially since info.Name() is readily available.

IOW
```go
if info.Name() != CgroupProcesses {
    return nil
}

Reply:
OK I haven't realized that
 - you've reused the GetAllPids code, inheriting all its issues
 - those issues (except for using WalkDir() instead of Walk() for go 1.16+) are already fixed by commit 5542a2c77d67605fa0fc9b562dfe8b3d19671b94